tasks/pre-install.yml

```diff
@@ -0,0 +1,38 @@
+--- 
+
+  # These tasks need to run before any other roles, to setup proper 
+  # environment variables depending on we're deploying to production, 
+  # staging, or development/test purposes. 
+  
+  # TODO: Perhaps this logic can be made more pretty somehow. 
+
+  - fail: msg="You must specify '-e deployment_environment=<production|staging|devel>'"
```

Comment:
This same forceful behaviour can be achieved by using inventory files, which allows the variables to be set there.
It seperates it a bit better rather than writing all three instances in the same file.